Tests/GitHubRepositories.tests.ps1

```diff
@@ -829,970 +741,1060 @@ try
+        It "Should return the correct type and properties for $_" {
+            $permissions.PSObject.TypeNames[0] | Should -Be 'GitHub.RepositoryActionsPermission'
+
+            $permissions.RepositoryName | Should -Be $repoName
+            $permissions.RepositoryUrl | Should -Be $repo.svn_url
+
+            if ($_ -eq 'Disabled')
+            {
+                $permissions.Enabled | Should -BeFalse
+            }
+            else
             {
-                $repo | Remove-GitHubRepository -Force
+                $permissions.Enabled | Should -BeTrue
+                $permissions.AllowedActions | Should -Be $_
             }
         }
     }
 
-    Describe 'GitHubRepositories\Set-GitHubRepositoryActionsPermission' {
+    Context "When specifiying the 'URI' Parameter from the Pipeline" {
         BeforeAll {
-            $repo = New-GitHubRepository -RepositoryName ([Guid]::NewGuid().Guid)
+            $permissions = $repo | Get-GitHubRepositoryActionsPermission
+        }
 
-            $allowedActions = 'All', 'LocalOnly', 'Selected', 'Disabled'
+        It 'Should return an object of the correct type' {
+            $permissions.PSObject.TypeNames[0] | Should -Be 'GitHub.RepositoryActionsPermission'
         }
+    }
 
-        foreach ($allowedAction in $allowedActions)
+    AfterAll {
+        if (Get-Variable -Name repo -ErrorAction SilentlyContinue)
         {
-            Context "When the AllowedAction Parameter is $allowedAction" {
-                BeforeAll {
-                    $setGitHubRepositoryActionsPermissionParms = @{
-                        Uri = $repo.svn_url
-                        AllowedActions = $allowedAction
-                    }
-
-                    Set-GitHubRepositoryActionsPermission @setGitHubRepositoryActionsPermissionParms
-                }
-
-                It 'Should have set the expected permissions' {
-                    $permissions = Get-GitHubRepositoryActionsPermission -Uri $repo.svn_url
-
-                    if ($allowedAction -eq 'Disabled')
-                    {
-                        $permissions.Enabled | Should -BeFalse
-                    }
-                    else
-                    {
-                        $permissions.Enabled | Should -BeTrue
-                        $permissions.AllowedActions | Should -Be $allowedAction
-                    }
-                }
-            }
+            $repo | Remove-GitHubRepository -Force
         }
+    }
+}
 
-        Context "When specifiying the 'URI' Parameter from the Pipeline" {
-            It 'Should not throw' {
-                { $repo | Set-GitHubRepositoryActionsPermission -AllowedActions 'All' } |
-                    Should -Not -Throw
+Describe 'GitHubRepositories\Set-GitHubRepositoryActionsPermission' {
+    BeforeAll {
+        $repo = New-GitHubRepository -RepositoryName ([Guid]::NewGuid().Guid)
+
+        $allowedActions = 'All', 'LocalOnly', 'Selected', 'Disabled'
```

Comment:
```suggestion
```

Looks like we either drop that, or reference it at 1446 (I think dropping it is fine).